### DISCUSSION TITLE
Bug in localization?

### DISCUSSION THREAD

**Discussion by svenbaus** [Megabit/Blazorise#4096] [Q&A]
Hello,
I'm using the ITextLocalizer and I'm facing a strange behaviour, maybe a bug. First my setup:

![grafik](https://user-images.githubusercontent.com/86463105/189609111-78d7a14b-c280-4fab-bbd6-1fce7f8a1789.png)

![grafik](https://user-images.githubusercontent.com/86463105/189609501-fe3fa382-e06d-450a-a493-da37d2c37585.png)

![grafik](https://user-images.githubusercontent.com/86463105/189611038-f45005b8-1c30-4f17-8fbb-ed7d9795c5c3.png)

I get the localized resourced for `ITextLocalizer<ProcessStepMessage> _processStepMessageLocalizer` in my object `ITextLocalizer<ProcessStep> _processStepLocalizer`. So I think the search for the resource has a bug or what am I doing wrong?

**Comment by stsrki** [comment]
From the screenshots, I cannot find what might be the cause of the problem. Is there any chance you could prepare us a small project to reproduce the issue on our side?

**Reply by svenbaus** [reply]
That will take some time. Is there some kind of tracing I can use to see, where the problem is?

**Reply by svenbaus** [reply]
So I've been able to get a small example running that demonstrates the problem. Just unzip it, run it and navigate to "/process/details/4c6ab2d9-67b3-4f1f-8ef7-30b783561cd2" (or any other guid), the data are hardcoded. Afterwards select the step and you will see my problem.

[example.zip](https://github.com/Megabit/Blazorise/files/9548377/example.zip)

**Reply by stsrki** [reply]
I found the problem. We have the wrong check `Contains( $"Resources.Localization.{resourceName}" )` so when there are multiple folders that starts the same all of them will be retrieved.

```diff
protected virtual string[] GetLocalizationResourceNames( Assembly assembly, string resourceName )
{
  return assembly.GetManifestResourceNames()
+    .Where( r => r.Contains( $"Resources.Localization.{resourceName}" ) && r.EndsWith( ".json" ) )
    .ToArray();
}
```

I will convert this to issue now.

**Reply by svenbaus** [reply]
Ok, thanks :). Is there any idea, in which release this bug will be resolved?

**Reply by stsrki** [reply]
It will be in 1.0.7. In the next week or two, I would guess.

**Reply by svenbaus** [reply]
Perfect, thanks for pointing out :).
